### PR TITLE
Add monotone spline options to utilties.

### PR DESCRIPTION
**Pull request by MFraters**
This add an algorithm which can produce monotone splines. The code was based on the javascript implementation on https://en.wikipedia.org/wiki/Monotone_cubic_interpolation.

### CODE REVIEW

**Comment by MFraters** [comment]
@gassmoeller I hope this adresses all your comments, or do you want me to restyle the whole spline code? 
I will try do design a good test later.

**Comment by MFraters** [comment]
I have now changed the spacing in the whole class anyway. Is this the kind of spacing you where looking for? I have also added a test to test this whole spline class.

File: include/aspect/utilities.h [hunk 1]
```diff
@@ -280,7 +280,8 @@ namespace aspect
            */
           void set_points(const std::vector<double> &x,
                           const std::vector<double> &y,
-                          bool cubic_spline = true);
+                          bool cubic_spline = true,
+                          bool monotome_spline = false);
```
Comment (tjhei) [inline include/aspect/utilities.h]: typo: "monotone"

File: source/utilities.cc [hunk 2]
```diff
@@ -638,209 +638,269 @@ namespace aspect
...
+                 * at https://en.wikipedia.org/wiki/Monotone_cubic_interpolation. The
+                 * parameters from this algorithm prevents overshooting in the
+                 * interpolation spline.
+                 */
+                std::vector<double> dys(n-1), dxs(n-1), ms(n-1);
+                for (unsigned int i=0; i < n-1; i++)
+                  {
+                    dxs[i] = x[i+1]-x[i];
+                    dys[i] = y[i+1]-y[i];
+                    ms[i] = double(dys[i])/double(dxs[i]);
```
Comment (tjhei) [inline source/utilities.cc]: these are doubles already, so remove the cast

**Comment by gassmoeller** [comment]
Thanks! This looks ready to merge except for the two minor things Timo mentioned. Lets see what the tester says.

**Comment by MFraters** [comment]
Any other comments? :)

File: include/aspect/utilities.h [hunk 3]
```diff
@@ -280,7 +280,8 @@ namespace aspect
            */
           void set_points(const std::vector<double> &x,
                           const std::vector<double> &y,
-                          bool cubic_spline = true);
+                          bool cubic_spline = true,
+                          bool monotone_spline = false);
```
Comment (bangerth) [inline include/aspect/utilities.h]: Can you document this additional argument? Also maybe in the class documentation?

It's not clear to me whether you can have both cubic and monotone splines. Is this possible?

Comment (bangerth) [inline include/aspect/utilities.h]: Oh, also mark both of the `bool` arguments as `const`.

File: source/utilities.cc [hunk 4]
```diff
@@ -638,209 +638,269 @@ namespace aspect
...
+            if (monotone_spline == true)
               {
-                A(i,i-1)=1.0/3.0*(x[i]-x[i-1]);
-                A(i,i)=2.0/3.0*(x[i+1]-x[i-1]);
-                A(i,i+1)=1.0/3.0*(x[i+1]-x[i]);
-                rhs[i]=(y[i+1]-y[i])/(x[i+1]-x[i]) - (y[i]-y[i-1])/(x[i]-x[i-1]);
+                /**
+                 * This monotone spline algorithm is based on the javascript version
+                 * at https://en.wikipedia.org/wiki/Monotone_cubic_interpolation. The
+                 * parameters from this algorithm prevents overshooting in the
```
Comment (bangerth) [inline source/utilities.cc]: prevents -> prevent

File: tests/splines.cc [hunk 5]
```diff
@@ -0,0 +1,55 @@
+/**
+ * This tests the Utilities::tk::spine class for the linear case,
```
Comment (bangerth) [inline tests/splines.cc]: spine -> spline

**Pull request merged**